contribute/design/1.x/csi/20190805-csi-zfspv-volume-provisioning.md

```diff
@@ -60,35 +60,35 @@ I should be able to delete the volume that was being consumed by my application.
 #### Volume Creation Workflow
 - CSI driver will handle CSI request for volume create
 - CSI driver will read the request parameters and create following resources:
-- ZfsVolume (Kubernetes custom resource)
-- ZfsVolume will be watched for the property change
+- ZFSVolume (Kubernetes custom resource)
+- ZFSVolume will be watched for the property change
 
 #### Volume Deletion Workflow
 - CSI driver will handle CSI request for volume delete
 - CSI driver will read the request parameters and delete corresponding ZFSPV volume resource
 
 #### High Level Design
 - user will setup all the node and setup the ZFS pool on each of those nodes.
-- user will deploy below sample storage class where we will get input of all the needed zfs properties. The storage class will have allowedTopologies and pool name which will tell us that pool is available on those nodes and it can pick any of that node to schedule the application pod.
+- user will deploy below sample storage class where we get all the needed zfs properties for creating the volume. The storage class will have allowedTopologies and pool name which will tell us that pool is available on those nodes and it can pick any of that node to schedule the PV. If allowed topologies are not provided then it means the pool is there on all the nodes and scheduler can create the PV anywhere.
 
 ```yaml
 kind: StorageClass
 apiVersion: storage.k8s.io/v1
 metadata:
   name: openebs-zfspv
-provisioner: openebs.io/zfs
+provisioner: zfs.csi.openebs.io
 parameters:
   blocksize: "4k"
   compression: "on"
   dedup: "on"
   thinprovision: "yes"
   poolname: "zfspv-pool"
-volumeBindingMode: WaitForFirstConsumer
```

Comment:
In the example, it is probably safer to retain this .. or atleast keep it with a comment.

Reply:
if WaitForFirstConsumer is used then kubernetes scheduler will take over,  so it's better to not mention it in OEP. I have thought of documenting it in README.